### DISCUSSION TITLE
Stock splits

### DISCUSSION THREAD

**Discussion by derekgair** [ghostfolio/ghostfolio#1072] [Q&A]
How does ghostfolio support stock splits, such as the recent 1:20 stock split of Amazon?  

I don't see anyway to enter one.

**Comment by dtslvr** [comment]
Hi @derekgair

Unfortunately, stock splits are not yet automated in Ghostfolio.
You can reimport historical data via _Admin Control_ > _Market Data_ > `AMZN` > `Gather Historical Market Data` to normalize the data, though. In a second step, go through each of your activities and multiply the quantity with 20 and divide the unit price by 20.

**Reply by derekgair** [reply]
I wonder if it is possible to create a new transaction type "split".   When iterating through the transactions in date order, the total quantity would be multiplied by the split amount (similar to the way buy/sell modifies the total quantity).  I assume the unit price would already have been updated by the price provider.

I've just cloned the repo, maybe I give the code change a try... When I get time :).

**Reply by dtslvr** [reply]
Yes, introducing a new type `SPLIT` sounds reasonable.
Let me know if I can support you with the setup.

**Reply by aaronspruit** [reply]
Oh man, after going through 4+ years of transactions, I would _really_ enjoy having a split capability.  So many manual adjustments, to have everything line up properly for a portfolio, which will only get worse with more history...

**Reply by Fedron** [reply]
I've mocked up a draft PR that adds a new `SPLIT` order type. Currently it's quite simple and doesn't support a mixed stock split.

@dtslvr I would appreciate if you could have a look at the PR to see if it's along the right lines? Then mixed stock split support and maybe getting stock split data automatically from Yahoo could be added later as improvement.

#3211

**Reply by anshuvajpayee** [reply]
Looks this functionality still not added

**Comment by Rwarren873** [comment]
Hate to dredge up an old conversation, but has anyone worked out a solution for this?  I have a couple of stocks that have had splits (normal and reverse) that I need to enter.

**Comment by bats77** [comment]
Indeed, it lacks a "type" category about merge&split like : 
1 Google = 100 (new Google) at date 
/ or / 
19 At&t at date = 15 (new)At&t + 3 WarnerMedia + cash (in bonus or malus).

Yahoo finance holds this kind of data ;-)

Nice job by the way, pursue 👍

**Comment by fferrin** [comment]
Although it's not a solution, a workaround I found for a stock split is to import the remaining shares at price `0` wth the `buy` type. I downloaded my report from Revolut and they report it that way


| Date       | Code  | DataSource | Currency | Price | Quantity | Action | Fee | Note |
|------------|-------|------------|----------|-------|----------|--------|-----|------|
| 2023-12-30 | STONK | YAHOO      | USD      | 100   | 1        | buy    | 0   |      |
| 2023-12-31 | STONK | YAHOO      | USD      | 0     | 9        | buy    | 0   | Stock split |
| 2023-12-31 | STONK | YAHOO      | USD      | 10    | 10       | sell   | 0   |      |

**Reply by richard-lee-863** [reply]
I have also done this, was hoping there was a better solution in the works. Would be nice to be able to see them as a different thing, so that it is filterable too.

**Comment by TheCaffeintedSloth** [comment]
Any movement on this / timeline? This is what I'd consider a rather large blocker for tracking stocks accurately.

**Reply by Kampvognen** [reply]
I second this feature request - with a long transaction list, this is essential. 
Is data being exposed through the Yahoo Finance API? https://finance.yahoo.com/calendar/splits/

**Reply by augustocollerone** [reply]
I would love to have this feature added. I've imported transactions dating a few years back and it's quite a mess to manually update the quantities to reflect the correct balance after the split

**Comment by kmanan** [comment]
Another vote for this feature. The limitation in the Action Type is making importing data and matching the numbers very difficult

**Comment by Hemie143** [comment]
Upvoting. Without this feature, it's a no-go for me to use this nice tool.

**Comment by sky-ermesombarros** [comment]
I've being spending some time to familiarise to the codebase to raise some proposal for this feature.
I can see that we already have a draft pr that adds the type SPLIT to enable stock splits https://github.com/ghostfolio/ghostfolio/pull/3211

But seems that it's a manual way to add that, and I was thinking on something more automatic.
I thought that we could have the splits imported by `yahoo-finance2` lib, but looking at their codebase seems they doesn't support that.

So, in order to provide something here, I will need to create a PR to add that type of information to `yahoo-finance2` and then import it into ghostfolio.

Will try to spend some time to create some drafts about my idea, but guess the best way to solve this is by importing the splits as we do for dividends and quotes.

**Reply by dtslvr** [reply]
Thanks for diving into the codebase, @sky-ermesombarros! Your idea to import stock splits automatically sounds great. However, I'd suggest starting with the core functionality.

How about focusing on the stock split activity first without any GUI or automation to keep it simple and ensure it works smoothly?

Looking forward to seeing your drafts!

**Reply by sky-ermesombarros** [reply]
That's a good idea.
TBH I'm focused more on modifyng the lib to get the information, then I will raise a draft with the modificaitons to do what you've suggested to then integrate it with the lib change.

**Comment by AlbertoGnz** [comment]
I would love to have this feature! is 100% needed for stock market. Thanks!!

**Comment by tvirus97** [comment]
Is there any update on this feature? Is it planned?

**Comment by amrit-venkatesh** [comment]
Upvoting this feature request! Stock splits are pretty common across markets. Ghostfolio really needs to sport this functionality please!

**Comment by Almadual** [comment]
Upvoting this feature request as well, needed!

**Comment by l3all** [comment]
Upvoting, Needed!

**Comment by TWV-GIT** [comment]
New year, new upvote

**Comment by AlexNathan345** [comment]
Upvoting, Needed!

**Comment by puace** [comment]
This data is informed by yahoo finance in the historical data, you can easily do a completely automatic portfolio update using ideas like python portfolyahoo (https://github.com/adriengivry/portfolyahoo).

Like somebody said, Ghostfolio, even being so good, is useless without some automatic split, reverse split and dividend portfolio automatic update, every month I get some split and my portfolio go completely messed...

Thank you!

**Comment by Patrickt21** [comment]
Picking this back up, building on #3211's design and @dtslvr's suggestion above to start with the core functionality (no GUI, no automatic import) before anything else.

Split into three independent PRs so each is easy to review on its own:

1. https://github.com/ghostfolio/ghostfolio/pull/7433 — adds the dormant `SPLIT` value to the `Type` enum + migration
2. _(to follow)_ allows a negative `quantity` for `SPLIT` in the create/update order DTO validation (positive = forward split ratio, negative = reverse split ratio), still rejecting `0`
3. _(to follow)_ makes the portfolio calculator apply `SPLIT` activities to the running share quantity in chronological order, without touching cost basis, invested amount or cash

Starting with PR 1 for feedback on the approach before continuing with 2 and 3.

**Comment by dtslvr** [comment]
We now have a dedicated issue tracking this feature: #7530

**Comment by dtslvr** [comment]
The stock splits feature implemented by @DavidReque is now live in Ghostfolio [3.52.0](https://github.com/ghostfolio/ghostfolio/releases/tag/3.52.0). Any feedback would be much appreciated!
